src/platform/Linux/WiFiPAFDriverCtrlIface.cpp

```diff
@@ -0,0 +1,524 @@
+    uint8_t peer_addr[6];
+    sscanf(peer_addr_str.c_str(), "%hhx:%hhx:%hhx:%hhx:%hhx:%hhx", &peer_addr[0], &peer_addr[1], &peer_addr[2], &peer_addr[3],
+           &peer_addr[4], &peer_addr[5]);
+
+    std::string ssi_str = reply_info["ssi"];
+    size_t ssi_buf_len  = ssi_str.size() / 2;
+    std::vector<uint8_t> ssi_buf;
+    ssi_buf.resize(ssi_buf_len);
+    if (chip::Encoding::HexToBytes(ssi_str.c_str(), ssi_str.size(), ssi_buf.data(), ssi_buf_len) != ssi_buf_len)
+    {
+        ChipLogError(DeviceLayer, "WiFi-PAF: error converting SSI to byte array");
+        return;
+    }
+    auto pPublishSSI = reinterpret_cast<const PAFPublishSSI *>(ssi_buf.data());
+
+    std::array<uint8_t, 6> peer_addr_array = { peer_addr[0], peer_addr[1], peer_addr[2], peer_addr[3], peer_addr[4], peer_addr[5] };
+
+    _OnReplied(peer_subscribe_id, publish_id, srv_proto_type, peer_addr_array, pPublishSSI);
+}
+
+void WiFiPAFDriverCtrlIface::OnNanReceive(std::unordered_map<std::string, std::string> received_data)
+{
+    ChipLogProgress(Controller, "WiFi-PAF: OnNanReceive");
+
+    WiFiPAF::WiFiPAFSession RxInfo;
+    ReturnOnFailure(StringToUInt32(received_data["id"], RxInfo.id));
+    ReturnOnFailure(StringToUInt32(received_data["peer_instance_id"], RxInfo.peer_id));
+
+    std::string peer_addr_str = received_data["address"];
+    sscanf(peer_addr_str.c_str(), "%hhx:%hhx:%hhx:%hhx:%hhx:%hhx", &RxInfo.peer_addr[0], &RxInfo.peer_addr[1], &RxInfo.peer_addr[2],
+           &RxInfo.peer_addr[3], &RxInfo.peer_addr[4], &RxInfo.peer_addr[5]);
+
+    std::string ssi_str = received_data["ssi"];
+    size_t ssi_buf_len  = ssi_str.size() / 2;
+    std::vector<uint8_t> ssi_buf;
+    ssi_buf.resize(ssi_buf_len);
+    if (chip::Encoding::HexToBytes(ssi_str.c_str(), ssi_str.size(), ssi_buf.data(), ssi_buf_len) != ssi_buf_len)
+    {
+        ChipLogError(DeviceLayer, "WiFi-PAF: error converting SSI to byte array");
+        return;
+    }
+
+    System::PacketBufferHandle buf = System::PacketBufferHandle::NewWithData(ssi_buf.data(), ssi_buf_len);
+
+    _OnNanReceive(RxInfo, std::move(buf));
+}
+
+void WiFiPAFDriverCtrlIface::OnNanPublishTerminated(std::unordered_map<std::string, std::string> publish_terminated_data)
+{
+    uint32_t publish_id;
+    ReturnOnFailure(StringToUInt32(publish_terminated_data["publish_id"], publish_id));
+
+    _OnNanPublishTerminated(publish_id, publish_terminated_data["reason"].c_str());
+}
+
+void WiFiPAFDriverCtrlIface::OnNanSubscribeTerminated(std::unordered_map<std::string, std::string> subscribe_terminated_data)
+{
+    uint32_t subscribe_id;
+    ReturnOnFailure(StringToUInt32(subscribe_terminated_data["subscribe_id"], subscribe_id));
+
+    _OnNanSubscribeTerminated(subscribe_id, subscribe_terminated_data["reason"].c_str());
+}
+
+CHIP_ERROR WiFiPAFDriverCtrlIface::Subscribe(const uint16_t & connDiscriminator, uint16_t mApFreq)
+{
+    ChipLogProgress(Controller, "WiFi-PAF: Try to subscribe the NAN-USD devices");
```

Comment:
Please use a consistent log category for the same component. DeviceLayer (as used below) seems more appropriate here than Controller.